pkg/apis/security/types_tokenrequest.go

```diff
@@ -0,0 +1,53 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package security
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// TokenRequest is resource that can be used to request WorkloadIdentity tokens.
+type TokenRequest struct {
+	metav1.TypeMeta
+	// Standard object metadata.
+	metav1.ObjectMeta
+	// Spec holds configuration settings for the requested token.
+	Spec TokenRequestSpec
+	// Status is bears the issued token with additional information back to the client.
+	Status TokenRequestStatus
+}
+
+// TokenRequestSpec holds configuration settings for the requested token.
+type TokenRequestSpec struct {
+	// ContextObject identify the object the token is requested for.
+	ContextObject *ContextObject
+	// Duration specifies for how long the requested token to be valid.
+	Duration *metav1.Duration
+}
+
+// ContextObject identify the object the token is requested for.
+type ContextObject struct {
+	// Kind of the object the token is requested for. Valid kinds are 'Shoot', 'Seed', etc.
+	Kind string
+	// API version of the object the token is requested for.
+	APIVersion string
+	// Name of the object the token is requested for.
+	Name string
+	// Namespace of the object the token is requested for.
+	Namespace string
```

Comment:
Do we need namespace here? Isn't this always the same namespace as the related `WorkloadIdentity`?
If we don't need it, maybe we can use https://github.com/kubernetes/api/blob/664bdd58a30d9e428a31e7e77713a26424025722/authentication/v1/types.go#L189?

Reply:
Yes, we need it. One example, the shoot `foo` from the namespace `garden-foo` is using the workload identity `bar` from the namespace `garden-bar` (`CredentialsBinding` can refer to Secrets/WorkloadIdentities across namespaces), then the shoot namespace != workload identity namespace.